### PR TITLE
Add exclude components flag

### PR DESCRIPTION
**Pull request by amitaekbote**

### CODE REVIEW

**Comment by amitaekbote** [comment]
@branden @mnaboka can I get a review please?

File: cmd/checks/components/components_check.go [hunk 1]
```diff
@@ -69,6 +70,7 @@ func Add(root *cobra.Command) {
 	componentsCmd.Flags().StringVarP(&healthURLPrefix, "health-url", "u", "/system/health/v1", "Set dcos-diagnostics health url")
 	componentsCmd.Flags().StringVarP(&scheme, "scheme", "s", "http", "Set dcos-diagnostics health url scheme")
 	componentsCmd.Flags().IntVarP(&port, "port", "p", 1050, "Set TCP port")
+	componentsCmd.Flags().StringArrayVar(&excludeComponents, "exclude-components", nil, "Exclude components from health check")
```
Comment (initcore0) [inline cmd/checks/components/components_check.go]: i think you need to use `StringArrayVarP`

File: cmd/checks/components/diagnostics_response.go [hunk 2]
```diff
@@ -18,10 +17,15 @@ type diagnosticsResponse struct {
 	} `json:"units"`
 }
 
-func (d *diagnosticsResponse) checkHealth() ([]string, int) {
+func (d *diagnosticsResponse) checkHealth(args []string) ([]string, int) {
 	var errorList []string
+	skipcomp := make(map[string]bool)
```
Comment (initcore0) [inline cmd/checks/components/diagnostics_response.go]: just a small improvement:
```go
skipComp := make(map[string]bool, len(args))
```

**Review by initcore0 — COMMENTED** [review]
few not, otherwise looks good

File: cmd/checks/components/components_check.go (hunk 1 above)
Comment (amitaekbote) [reply cmd/checks/components/components_check.go]: https://godoc.org/github.com/spf13/pflag#StringArrayVar
The difference is a shorthand which I wasnt planning on using.

**Comment by amitaekbote** [comment]
@mnaboka @branden can one of you review this please?

File: cmd/checks/components/components_check_test.go [hunk 3]
```diff
@@ -82,8 +82,15 @@ func TestDiagnosticsResponse(t *testing.T) {
...
+
+	complist := []string{"dcos-checks-poststart.service", "dcos-checks-poststart.timer"}
+	_, retCode := dr.checkHealth(complist)
 	if retCode != 0 {
 		t.Fatalf("Check health failed")
 	}
+
+	_, retCode = dr.checkHealth(nil)
+	if retCode == 0 {
+		t.Fatalf("Check health failed")
```
Comment (branden) [inline cmd/checks/components/components_check_test.go]: This should have a different error message from the above stanza. The above expects checks to pass, this expects checks to fail.

File: cmd/checks/components/components_check.go (hunk 1 above)
Comment (branden) [inline cmd/checks/components/components_check.go]: nit: `--exclude-components` is pretty wordy; `components` is already there as the subcommand. `--exclude` or `--skip` is enough to get the idea across I think.

**Review by branden — CHANGES_REQUESTED** [review]
The new test needs a clearer error message. One nit on the name of the flag but otherwise LGTM.

File: cmd/checks/components/diagnostics_response.go [hunk 4]
```diff
@@ -18,10 +17,15 @@ type diagnosticsResponse struct {
 	} `json:"units"`
 }
 
-func (d *diagnosticsResponse) checkHealth() ([]string, int) {
+func (d *diagnosticsResponse) checkHealth(args []string) ([]string, int) {
```
Comment (branden) [inline cmd/checks/components/diagnostics_response.go]: This arg needs a more descriptive name than `args`.

**Review by branden — CHANGES_REQUESTED** [review]

**Comment by amitaekbote** [comment]
@branden @mnaboka ready for review. Tested with https://github.com/dcos/dcos/pull/2069

File: cmd/checks/components/components_check.go [hunk 5]
```diff
@@ -69,6 +70,7 @@ func Add(root *cobra.Command) {
 	componentsCmd.Flags().StringVarP(&healthURLPrefix, "health-url", "u", "/system/health/v1", "Set dcos-diagnostics health url")
 	componentsCmd.Flags().StringVarP(&scheme, "scheme", "s", "http", "Set dcos-diagnostics health url scheme")
 	componentsCmd.Flags().IntVarP(&port, "port", "p", 1050, "Set TCP port")
+	componentsCmd.Flags().StringArrayVar(&excludeComponents, "exclude", nil, "Exclude components from health check")
```
Comment (initcore0) [inline cmd/checks/components/components_check.go]: i'd still want the options to be consistent, it should be `StringArrayVarP` as the rest of the flags, or you need to refactor the rest without `P`

Comment (amitaekbote) [reply cmd/checks/components/components_check.go]: done

**Review by initcore0 — APPROVED** [review]

**Review by branden — APPROVED** [review]

**Pull request merged**